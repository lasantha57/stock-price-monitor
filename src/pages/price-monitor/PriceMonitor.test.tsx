import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import PriceMonitor from './index';

afterEach(cleanup)

describe('Testing PriceMonitor', () => {
    it('Component should render correctly', () => {
        const tree = renderer
            .create(
                <Router>
                    <PriceMonitor />
                </Router>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});