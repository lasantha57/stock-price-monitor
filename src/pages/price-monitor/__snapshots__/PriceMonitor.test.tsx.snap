// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PriceMonitor Component should render correctly 1`] = `
<div
  className="MuiContainer-root css-ur2jdm-MuiContainer-root"
>
  <div
    className="MuiBox-root css-rm9e7r"
  >
    <div
      className="MuiBox-root css-1ebprri"
    >
      <div
        className="MuiFormControl-root css-10n666l-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
        >
          Price Source *
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
          required={true}
          variant="outlined"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-source"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-component-select-source"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Please select Price Source
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="source"
            onAnimationStart={[Function]}
            onChange={[Function]}
            placeholder="Source"
            required={true}
            tabIndex={-1}
            value={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1in441m"
            >
              <span>
                Price Source
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-1ebprri"
    >
      <div
        className="MuiFormControl-root css-10n666l-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
        >
          Ticker *
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
          required={true}
          variant="outlined"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-ticker"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-component-select-ticker"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Please select Ticker
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="ticker"
            onAnimationStart={[Function]}
            onChange={[Function]}
            placeholder="Ticker"
            required={true}
            tabIndex={-1}
            value={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1in441m"
            >
              <span>
                Ticker
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-5vb4lz"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="A price table"
          className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15bsjcf-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Time
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Price
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
    </div>
  </div>
</div>
`;
